tests/unit/test_spaces.cpp

```diff
@@ -0,0 +1,122 @@
+#include "gtest/gtest.h"
+#include "VecSim/spaces/space_aux.h"
+#include "VecSim/spaces/IP/IP.h"
+#include "VecSim/spaces/IP/IP_SSE.h"
+#include "VecSim/spaces/IP/IP_AVX.h"
+#include "VecSim/spaces/IP/IP_AVX512.h"
+#include "VecSim/spaces/L2/L2.h"
+#include "VecSim/spaces/L2/L2_SSE.h"
+#include "VecSim/spaces/L2/L2_AVX.h"
+#include "VecSim/spaces/L2/L2_AVX512.h"
+
+class SpacesTest : public ::testing::Test {
+
+protected:
+    SpacesTest() {}
+
+    ~SpacesTest() override {}
+
+    void SetUp() override {}
+
+    void TearDown() override {}
+};
+
+// This test will trigger the "Residuals" function for dimension > 16, for each optimization.
+TEST_F(SpacesTest, l2_17) {
+    Arch_Optimization optimization = getArchitectureOptimization();
```

Comment:
We guarantee that `optimization` is `ARCH_OPT_AVX512` in CI?

Reply:
not for now. I'll explain f2f